### DISCUSSION TITLE
Is it possible to generate SSL + Config files for servers outside the docker environment?

### DISCUSSION THREAD

**Discussion by UKIkarus** [nginx-proxy/nginx-proxy#1660] [Q&A]
Having a bit of trouble here, odd request maybe.

What I have is a docker environment running this which works fine for all the containers, no issues, what I also have is a service on another physical/virtual host on a totally independent IP within the network.

Can I have this generate an SSL certificate and associated config entry to redirect testsite1.example.com to https://10.0.52.101 automatically with the proxy pass etc?

If so, where would I begin to define this so that the nginx-letsencrypt and Docker-Gen will generate the necessary entries from the templates?

I don't mind creating dummy containers if it achieves the desired goal.

I believe I can use the Standalone certificates feature to create the certificates if I run an instance of the acme-companion with its own standalone certs configuration, fine, if that is what is required I'll do some digging.

Where do I define a custom entry for this server so that it effectively includes it into the default.conf for nginx though? so I can add my own entries as docker-gen will naturally overwrite the files.

**Comment by UKIkarus** [comment]
Seriously? not possible? not a single response?

**Reply by buchdag** [reply]
Sorry for the delayed answer.

Unfortunately no, that's not possible to proxy to something else than a Docker container at the moment, `nginx-proxy` was really only made with Docker containers in mind and was never meant to be a general purpose reverse proxy.

While this feature might be added some day, `nginx-proxy` has gone a long period with very low to no maintenance and features that are far from the original scope like this one (even if they might be very useful) are low priority.

**Reply by buchdag** [reply]
I'm transferring this to `nginx-proxy/nginx-proxy` because that's more relevant to this repo.

**Comment by UKIkarus** [comment]
Damn it, ok, pity as the setup is so useful with automatic certs and proxying it would just be great to add other manual entries to what the nginx-gen does somehow.

________________________________

**Comment by tkw1536** [comment]
This has been requested before, see issue #998. 

It's not a feature as-is but there are several workarounds in the issue above.

**Reply by buchdag** [reply]
@tkw1536 I wasn't certain that would work with custom configuration file, if it does that's another argument for not putting too much effort into this feature right now, but describing a proven workaround in the doc would be an excellent idea.

**Reply by tkw1536** [reply]
I was mostly referring to https://github.com/nginx-proxy/nginx-proxy/issues/998#issuecomment-366953378 - Having a docker container that "proxies" to the proper host containing the actual application to be served. This allows leaving the docker environment. 

I've used this approach in production for several years now -- (ab)using `nginx-proxy` only as an easy to configure ingress controller which terminates SSL. All application servers are handled entirely separately.

**Comment by UKIkarus** [comment]
In this instance it's not the same host running docker it's an entirely different physical machine on the network I want to proxy to.

________________________________
